services/order.py

```diff
@@ -0,0 +1,61 @@
+from datetime import datetime
+from typing import Any
+
+from django.core.exceptions import ValidationError
+from django.db import transaction
+from django.db.models import QuerySet
+
+from db.models import Order, Ticket, User, MovieSession
+
+
+def create_order(
+    tickets: list[dict[str, Any]],
+    username: str,
+    date: datetime.date = None,
+) -> Order:
+    try:
+        user = User.objects.get(username=username)
+    except User.DoesNotExist:
+        raise ValidationError(
+            f"User with username '{username}' does not exist."
+        )
+
+    with transaction.atomic():
+        order = Order.objects.create(user=user)
+
+        if date:
+            created_at = datetime.strptime(date, "%Y-%m-%d %H:%M")

```

Comment:
Consider adding error handling for the `strptime` method to catch any `ValueError` exceptions that may arise from an incorrect date format. This will ensure that the date is in the correct format and prevent potential errors.